### PR TITLE
Check syntax of stitching detailed

### PR DESCRIPTION
**Pull request by LaurentBerger**
access to base class in python

### CODE REVIEW

**Comment by LaurentBerger** [comment]
PR not ready I want to add command line in tutorial. Is it a good idea? Can I add some lines in https://docs.opencv.org/trunk/d0/d33/tutorial_table_of_content_stitching.html  and create a small tuto in tutorial_py_root ?

#    sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/s1.jpg','F:/lib/opencv_extra/testdata/stitching/s2.jpg',
#           '--work_megapix', '0.6','--features','surf','--matcher','affine','--estimator','affine',
#            '--match_conf','0.3','--conf_thresh','0.3','--ba','affine','--ba_refine_mask','xxxxx',
#            '--wave_correct','no','--warp','affine','--blend','multiband']
#sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/s1.jpg','F:/lib/opencv_extra/testdata/stitching/s2.jpg',
#            '--work_megapix', '0.6','--features','surf','--matcher','homography','--estimator','homography',
#            '--match_conf','0.1','--conf_thresh','0.3','--ba','ray','--ba_refine_mask','xxxxx',
#            '--wave_correct','no','--warp','fisheye','--blend','multiband']
#sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/a1.png','F:/lib/opencv_extra/testdata/stitching/a2.png','F:/lib/opencv_extra/testdata/stitching/a3.png',
#            '--work_megapix', '0.6','--features','orb','--matcher','homography','--estimator','homography',
#            '--match_conf','0.1','--conf_thresh','0.3','--ba','reproj','--ba_refine_mask','xxxxx',
#            '--wave_correct','no','--warp','cylindrical','--blend','feather']
sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/boat1.jpg','F:/lib/opencv_extra/testdata/stitching/boat2.jpg','F:/lib/opencv_extra/testdata/stitching/boat3.jpg',
            '--work_megapix', '0.6','--features','orb','--matcher','homography','--estimator','homography',
            '--match_conf','0.1','--conf_thresh','0.3','--ba','ray','--ba_refine_mask','xxxxx','--save_graph','test.txt',
            '--wave_correct','no','--warp','cylindrical','--blend','multiband','--expos_comp','no','--seam','gc_colorgrad']
#sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/a1.png','F:/lib/opencv_extra/testdata/stitching/a2.png','F:/lib/opencv_extra/testdata/stitching/a3.png',
#            '--work_megapix', '0.6','--features','orb','--matcher','homography','--estimator','homography',
#            '--match_conf','0.1','--conf_thresh','0.3','--ba','affine','--ba_refine_mask','xxxxx',
#            '--wave_correct','horiz','--warp','stereographic','--blend','feather']
#sys.argv = [sys.argv[0], 'F:/lib/opencv_extra/testdata/stitching/s1.jpg','F:/lib/opencv_extra/testdata/stitching/s2.jpg',
#        '--work_megapix', '0.6','--features','surf','--matcher','affine','--estimator','affine',
#        '--match_conf','0.3','--conf_thresh','0.3','--ba','affine','--ba_refine_mask','xxxxx',
#        '--wave_correct','no','--warp','affine','--timelapse','as_is']

**Comment by alalek** [comment]
Method `setNrFeeds()` is not available in ExposureCompensator **base** class.
But it is available for derived GainCompensator / ChannelsCompensator / BlocksCompensator.

<details>

<summary>Python "help(compensator)" statement</summary>

```
Help on detail_ExposureCompensator object:

class detail_ExposureCompensator(__builtin__.object)
 |  Methods defined here:
 |  
 |  __repr__(...)
 |      x.__repr__() <==> repr(x)
 |  
 |  apply(...)
 |      apply(index, corner, image, mask) -> image
 |      .   @brief Compensate exposure in the specified image.
 |      .   
 |      .   @param index Image index
 |      .   @param corner Image top-left corner
 |      .   @param image Image to process
 |      .   @param mask Image mask
 |  
 |  feed(...)
 |      feed(corners, images, masks) -> None
 |      .   @param corners Source image top-left corners
 |      .   @param images Source images
 |      .   @param masks Image masks to update (second value in pair specifies the value which should be used
 |      .   to detect where image is)
 |  
 |  getMatGains(...)
 |      getMatGains([, arg1]) -> arg1
 |      .
 |  
 |  getUpdateGain(...)
 |      getUpdateGain() -> retval
 |      .
 |  
 |  setMatGains(...)
 |      setMatGains(arg1) -> None
 |      .
 |  
 |  setUpdateGain(...)
 |      setUpdateGain(b) -> None
 |      .
 |  
 |  ----------------------------------------------------------------------
 |  Static methods defined here:
 |  
 |  createDefault(...)
 |      createDefault(type) -> retval
 |      .
 |  
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |  
 |  __new__ = <built-in method __new__ of type object>
 |      T.__new__(S, ...) -> a new object with type S, a subtype of T
```

</details>

**Comment by LaurentBerger** [comment]
Yes I'm agree but when createdefault is called : 

`compensator = cv.detail.ExposureCompensator_createDefault(cv.detail.ExposureCompensator_CHANNELS
`
I understand that a[ channelcompesator object](https://github.com/opencv/opencv/blob/master/modules/stitching/src/exposure_compensate.cpp#L52-L70) is return then 

`compensator.setNrFeeds(expos_comp_nr_feeds)`

should be good. Now I thnik it's same problem described in https://github.com/opencv/opencv/issues/13022

**Comment by LaurentBerger** [comment]
> 
> 
> OpenCV Python bindings doesn't manage inheritance in this way.
> `createDefault()` returns `Ptr<ExposureCompensator>` so it is wrapped into "ExposureCompensator" Python interface.

Can I remove CV_WRAP [here ](https://github.com/opencv/opencv/blob/master/modules/stitching/include/opencv2/stitching/detail/exposure_compensate.hpp#L67) ?

**Comment by alalek** [comment]
It is better to create compensator using of wrapper of exact derived class. This approach will provide access to specific methods/properties.

However this call may be used to specify "default" compensator for main stitching algorithm.

File: samples/python/stitching_detailed.py [hunk 1]
```diff
@@ -379,9 +394,14 @@
...
+        blender.feed(cv.UMat(image_warped_s), mask_warped, corners[idx])
 if not timelapse:
     result=None
     result_mask=None
     result,result_mask = blender.blend(result,result_mask)
-    cv.imwrite(result_name,result)
+    zoomx =600/result.shape[1]
+    dst=cv.normalize(src=result,dst=None,alpha=255.,norm_type=cv.NORM_MINMAX,dtype=cv.CV_8U)
+    dst=cv.resize(dst,dsize=None,fx=zoomx,fy=zoomx)
+    cv.imwrite(result_name,dst)
```
Comment (alalek) [inline samples/python/stitching_detailed.py]: I believe we should write into file non-resized (original) result.
For `imshow()`, resize is fine.

Comment (LaurentBerger) [reply samples/python/stitching_detailed.py]: OK

**Pull request merged**